src/project_file_walker.rs

```diff
@@ -1,52 +1,75 @@
+    },
 };
 
-trait IntoParallelIterator {
-    fn into_parallel_iterator(self) -> IterBridge<WalkParallelIterator>;
-}
-
-impl IntoParallelIterator for WalkParallel {
-    fn into_parallel_iterator(self) -> IterBridge<WalkParallelIterator> {
-        WalkParallelIterator::new(self).par_bridge()
-    }
+pub(crate) fn into_parallel_iterator(
+    walk_parallel: WalkParallel,
+) -> IterBridge<WalkParallelIterator> {
+    WalkParallelIterator::new(walk_parallel).par_bridge()
 }
 
 pub(crate) struct WalkParallelIterator {
-    receiver_iterator: <Receiver<DirEntry> as IntoIterator>::IntoIter,
+    receiver_iterator: <Receiver<(DirEntry, Vec<SupportedLanguage>)> as IntoIterator>::IntoIter,
     _handle: JoinHandle<()>,
 }
 
 impl WalkParallelIterator {
     pub fn new(walk_parallel: WalkParallel) -> Self {
-        let (sender, receiver) = mpsc::channel::<DirEntry>();
+        let (sender, receiver) = mpsc::channel::<(DirEntry, Vec<SupportedLanguage>)>();
         let handle = thread::spawn(move || {
+            let ignore = &walk_parallel.ignore();
             walk_parallel.run(move || {
                 Box::new({
                     let sender = sender.clone();
-                    move |entry| {
-                        let entry = match entry {
+                    move |entry_and_match_metadata| {
+                        let (entry, match_metadata) = match entry_and_match_metadata {
                             Err(err) => {
                                 err_message!("{err}");
                                 return WalkState::Continue;
                             }
-                            Ok(entry) => entry,
+                            Ok(entry_and_match_metadata) => entry_and_match_metadata,
                         };
                         if !entry.metadata().unwrap().is_file() {
                             return WalkState::Continue;
                         }
-                        sender.send(entry).unwrap();
+                        let matched_languages = match_metadata
+                            .or_else(|| {
+                                ignore
+                                    .should_skip_entry_with_match_metadata_token(&entry)
+                                    .1
+                                    .map(|match_metadata_token| {
+                                        ignore.get_match_metadata(match_metadata_token)
+                                    })
+                            })
+                            .map(|match_metadata| {
+                                match_metadata
+                                    .matching_file_types
+                                    .map(|file_type_def| {
+                                        ALL_SUPPORTED_LANGUAGES_BY_NAME_FOR_IGNORE_SELECT
+                                            .get(&file_type_def.name())
+                                            .copied()
+                                            .unwrap()
```

Comment:
Thinking of eg the Typescript `Record` type (with a union-of-string-literals as its key type), it seems like it might be nice to have it encoded at the type level that lookups into a "known/enumerated keys + fully filled on initialization" `HashMap` like this should never fail (so then consumer-wise you wouldn't `.unwrap()` after `.get()`'ing)?

Reply:
Hmm, yeah, I'm not sure how you'd achieve that exactly.